### PR TITLE
Change to Ubuntu 16.04 Xenial LTS

### PR DESCRIPTION
**Pull request by ptibbetts**
This PR is a work-in-progress, aiming to replace Ubuntu 14.04 Trusty with 16.04 Xenial. 

**Todo:**

- [x] confirm if remote-user role should change from ping to whoami
- [x] [wait for Vagrant 1.8.5 to be released](https://github.com/roots/trellis/pull/604#issuecomment-228943575)
- [x] switch Vagrant boxes to [`bento/ubuntu-16.04`](https://atlas.hashicorp.com/bento/boxes/ubuntu-16.04)
- [ ] add support for vmware_workstation
- [ ] add support for parallels

https://github.com/roots/trellis/pull/604/commits/9751f498265d28bee50085a58c8bae3c2a6bf552 is based on the method described in [this post on the Roots Discourse](https://discourse.roots.io/t/trellis-ubuntu-16-04-do/6551/3?u=paul_tibbetts) which

* changed the Vagrant box to `geerlingguy/ubuntu1604`, which ships with Python 2.7.11+
* changed the Ansible remote-user role from `ping`ing the external server to running a raw `whoami`. It also removes `cli_options_ping` from `vars.py` in the Trellis library.
* added [python-minimal](http://packages.ubuntu.com/xenial/python2.7-minimal) to the remote server using a raw command as the admin user. This is root on the first run of the playbook and if another admin user is chosen the command will run (and pass) as them in the future. [Debian describes](https://packages.debian.org/wheezy/python2.7-minimal) python-minimal as "… the interpreter and some essential modules. It can be used in the boot process for some basic tasks" which is all we need

### CODE REVIEW

**Comment by ptibbetts** [comment]
It would remove the need for PR https://github.com/roots/trellis/pull/596 as Ubuntu 16.04 ships with OpenSSL 1.0.2g-fips

```
vagrant@example:~$ openssl version
OpenSSL 1.0.2g-fips  1 Mar 2016
```

And has ALPN support:

* HTTP/2 Enabled: true
* SPDY/3.1 Enabled: false
* Use Alternative Service: false
* ALPN Protocols: h2,http/1.1
* NPN Protocols: undefined
* HTTP/2 sessions

Host | Proxy | ID | Protocol Negotiated | Active streams | Unclaimed pushed	| Max	Initiated	Pushed | Pushed and claimed | Abandoned	| Received frames |	Secure | Sent settings | Received settings | Send window | Receive window | Unacked received data | Error
--- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | 
178.62.33.18:443 | direct:// | 4918 | h2 | 0 | 0 | 128 | 12 | 0 | 0 | 0 | 34 | true | true | true | 2147483647 | 15728640 | 112022 | 0
clients4.google.com:443 |	direct:// |	5076 | h2	| 0 | 0 | 100 | 0 | 0 | 0 | 0 |	0 |	true | true | true | 1048576 | 15728640 | 0 | 0
secure.gravatar.com:443 |	direct:// |	5032 | h2	| 0 | 0 | 128 | 2 | 0 | 0 | 0 |	2 |	true | true | true | 2147483647 | 15728640 | 0 | 0
clients4.google.com:443 |	direct:// |	5061 | h2	| 0 | 0 | 100 | 0 | 0 | 0 | 0 |	0 |	true | true | true | 1048576 | 15728640 | 0 | 0

And makes the ⚡️ go blue using the [HTTP/2 and SPDY indicator Chrome extension](https://chrome.google.com/webstore/detail/http2-and-spdy-indicator/mpbpobfflnpcgagjijhmgnchggcjblin)

screen shot 2016-06-27 at 00 07 58

Whereas Ubuntu 14.04 doesn't get a ⚡️

screen shot 2016-06-27 at 00 07 39

**Comment by ptibbetts** [comment]
Just checking, the `cli_options_ping` var was only used to find out which user to connect wasn't it? Running a [raw](http://docs.ansible.com/ansible/raw_module.html) command returns a success if the command is passed in the same way that `ping` did. 

It wasn't used for anything else was it?

**Comment by ptibbetts** [comment]
This PR **also** assumes Roots wants to drop Ubuntu 14.04 and support one OS at a time, instead of adding support for choosing/automatically detecting which OS you're using.

**Comment by swalkinshaw** [comment]
cc @fullyint for the question about `cli_options_ping`.

@ptibbetts thanks for this. I had actually tested this myself a few days and got it working with `geerlingguy/ubuntu1604` as well.

All I had to change was adding `python` to common packages.

There's one issue that I realized with switching to 16.04 though. In our `Vagrantfile` we support `vmware_workstation` and `parallels` providers too. It would be nice to keep support for those if possible. They require different base boxes.

**Comment by ptibbetts** [comment]
> All I had to change was adding `python` to common packages.

I've rolled back, added python to the with_items list and it works fine with Vagrant but gives me the following on a DO server:

```
➜  trellis (feature/ubuntu1604) ✗ ansible-playbook server.yml -e env=staging -vvvv
Using /Users/paul/GitHub/trellis/ansible.cfg as config file
Loaded callback output of type stdout, v2.0

PLAYBOOK: server.yml ***********************************************************
3 plays in server.yml

PLAY [Ensure necessary variables are defined] **********************************

TASK [Ensure environment is defined] *******************************************
task path: /Users/paul/GitHub/trellis/variable-check.yml:8
skipping: [localhost] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

PLAY [Determine Remote User] ***************************************************

TASK [remote-user : Determine whether to connect as root or admin_user] ********
task path: /Users/paul/GitHub/trellis/roles/remote-user/tasks/main.yml:2
ESTABLISH LOCAL CONNECTION FOR USER: paul
localhost EXEC /bin/sh -c '( umask 22 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1467039055.03-78956122214624 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1467039055.03-78956122214624 `" )'
localhost PUT /var/folders/g4/db69vlp163s1fzhldrwdyh9m0000gn/T/tmpTbeKqv TO /Users/paul/.ansible/tmp/ansible-tmp-1467039055.03-78956122214624/command
localhost EXEC /bin/sh -c 'LANG=en_GB.UTF-8 LC_ALL=en_GB.UTF-8 LC_MESSAGES=en_GB.UTF-8 /usr/bin/python /Users/paul/.ansible/tmp/ansible-tmp-1467039055.03-78956122214624/command; rm -rf "/Users/paul/.ansible/tmp/ansible-tmp-1467039055.03-78956122214624/" > /dev/null 2>&1'
ok: [46.101.7.133 -> localhost] => {"changed": false, "cmd": ["ansible", "46.101.7.133", "-m", "ping", "-u", "root", "--connection=smart", "--timeout=10", "--inventory-file=hosts"], "delta": "0:00:00.782153", "end": "2016-06-27 15:50:55.897881", "failed": false, "failed_when_result": false, "invocation": {"module_args": {"_raw_params": "ansible 46.101.7.133 -m ping -u root --connection='smart' --timeout='10' --inventory-file='hosts'", "_uses_shell": false, "chdir": null, "creates": null, "executable": null, "removes": null, "warn": true}, "module_name": "command"}, "rc": 2, "start": "2016-06-27 15:50:55.115728", "stderr": "", "stdout": "\u001b[0;31m46.101.7.133 | FAILED! => {\n    \"changed\": false, \n    \"failed\": true, \n    \"module_stderr\": \"\", \n    \"module_stdout\": \"/bin/sh: 1: /usr/bin/python: not found\\r\\n\", \n    \"msg\": \"MODULE FAILURE\", \n    \"parsed\": false\n}\u001b[0m", "stdout_lines": ["\u001b[0;31m46.101.7.133 | FAILED! => {", "    \"changed\": false, ", "    \"failed\": true, ", "    \"module_stderr\": \"\", ", "    \"module_stdout\": \"/bin/sh: 1: /usr/bin/python: not found\\r\\n\", ", "    \"msg\": \"MODULE FAILURE\", ", "    \"parsed\": false", "}\u001b[0m"], "warnings": []}

TASK [remote-user : Set remote user for each host] *****************************
task path: /Users/paul/GitHub/trellis/roles/remote-user/tasks/main.yml:8
ok: [46.101.7.133] => {"ansible_facts": {"ansible_user": "admin"}, "changed": false, "invocation": {"module_args": {"ansible_user": "admin"}, "module_name": "set_fact"}}

TASK [remote-user : Announce which user was selected] **************************
task path: /Users/paul/GitHub/trellis/roles/remote-user/tasks/main.yml:12
Note: Ansible will attempt connections as user = admin
ok: [46.101.7.133] => {}

PLAY [WordPress Server - Install LEMP Stack with PHP 7.0 and MariaDB MySQL] ****

TASK [setup] *******************************************************************
<46.101.7.133> ESTABLISH SSH CONNECTION FOR USER: admin
<46.101.7.133> SSH: EXEC ssh -C -vvv -o ForwardAgent=yes -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=admin -o ConnectTimeout=10 -o ControlPath=/Users/paul/.ansible/cp/ansible-ssh-%h-%p-%r 46.101.7.133 '/bin/sh -c '"'"'( umask 22 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1467039056.13-64090783706207 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1467039056.13-64090783706207 `" )'"'"''
System info:
  Ansible 2.0.2.0; Darwin
  Trellis at "Upgraded Ubuntu from 14.04 Trusty to 16.04 Xenial"
---------------------------------------------------
Failed to connect to the host via ssh.
fatal: [46.101.7.133]: UNREACHABLE! => {"changed": false, "unreachable": true}
	to retry, use: --limit @server.retry

PLAY RECAP *********************************************************************
46.101.7.133               : ok=3    changed=0    unreachable=1    failed=0
localhost                  : ok=0    changed=0    unreachable=0    failed=0
```
which makes me think the `geerlingguy/ubunty1604` box has been patched to add the python package. I've tried using the `ubuntu/xenial64` box before but Ansible tries to connect with ubuntu as the user using a password so I quickly switched back to the geerlingguy version.

> In our Vagrantfile we support vmware_workstation and parallels providers too

I'd completely overlooked these as I don't use them,  I could give the parallels one a go using their 14 day trial but I'd have to ask someone else to help with the vmware one

**Comment by fullyint** [comment]
**Short**
I think you'd want to keep `cli_options_ping` but rename it to just `cli_options` (also rename in [users role](https://github.com/roots/trellis/blob/263d05561f1aa0b61ba5162cb71d752cdba54fb3/roles/users/tasks/main.yml#L54)).

**Long**
I understand how `cli_options_ping` could appear unnecessary after the switch to the `raw` module:
- Without the variable, a project with the Trellis defaults works. However, if users add options to the primary `ansible-playbook` command (e.g., a custom `--private-key`), the proposed ad hoc raw command would not receive the information and could inappropriately fail to connect.
- The substring 'ping' in the variable name implies that the options apply to just the ping module. However, the cli options would also apply to the connection made by any ansible ad hoc command. When I named the original variable, I failed to consider its use beyond just the `ping` module.

The var `cli_options_ping` was introduced in #578 to be sure that any critical options users could pass to the `ansible-playbook` command would be recorded and passed to the ad hoc ping command. As a `local_action`, the ad hoc ping was otherwise a completely separate connection, unaware of connection settings passed via the cli.

For example, suppose users pass `--inventory` because their inventory is in a custom location. The ad hoc raw command must also be informed of that location. Similarly, if users must pass options such as `--ask-pass` or `--private-key` to the primary `ansible-playbook` command, presumably the info of these options will be indispensable in testing a remote-user's ability to connect via the ad hoc raw command.

-----

**Testing**
This switch to the `raw` module should probably also test that it functions properly in ...
>the scenario that a failed connection for `root` could still give a return code of `0` [as] it does on an Ubuntu box on AWS ([ref](https://github.com/roots/trellis/issues/285#issuecomment-127082954))

The `ping` module gave the correct return code, so hopefully the `raw` module will too.

**Comment by ptibbetts** [comment]
Thanks @fullyint , I can see why `cli_options` definitely needs to stay. I'll also check it returns properly for AWS.

I've just tried appending the cli_options to the raw module and had no luck. I might have to rethink how to install python on the first run.

**Comment by fullyint** [comment]
@ptibbetts Thank you for your time and work on this.
You mentioned...
> I've just tried appending the cli_options to the raw module and had no luck.

What did you run into? Here's a [demo branch](https://github.com/fullyint/trellis/commit/49993b5f3db42167af94caf7dcec698f59fb6344) where `cli_options` seems to work for me.

**Return codes.** Whereas the `ping` module returns failure for `root` on AWS (when it worked on 14.04), the `raw` module apparently still returns `rc=0` for AWS when connecting as `root`, e.g.,
```
$ ansible ip.xxx.xxx.xx -m raw -a whoami -u root -i hosts/<environment>
ip.xxx.xxx.xx | SUCCESS | rc=0 >>
Please login as the user "ubuntu" rather than the user "root".
```
(I don't know much about AWS. I tested using the community AMI
`ubuntu/images/hvm-ssd/ubuntu-xenial-16.04-amd64-server-20160516.1 - ami-06b94666`.)

The remote-user role currently uses the return code in a ternary conditional to dynamically define `ansible_user`. It looks like the return code won't be reliable, so my demo branch changes the conditional from `root_status | success` to `('root' in root_status.stdout_lines)`. That just checks if the user is actually echoed to stdout on its own line, which is false in the AWS output above). There might be a smarter approach.

File: README.md [hunk 1]
```diff
@@ -12,13 +12,13 @@ Ansible playbooks for setting up a LEMP stack for WordPress.
...
 
-* Ubuntu 14.04 Trusty LTS
+* Ubuntu 16.04 Xenial LTS
 * Nginx (with optional FastCGI micro-caching)
 * PHP 7.0
 * MariaDB (a drop-in MySQL replacement)
 * SSL support (scores an A+ on the [Qualys SSL Labs Test](https://www.ssllabs.com/ssltest/))
 * Let's Encrypt integration for free SSL certificates
-* HTTP/2 support (requires SSL)
+* HTTP/2 + ALPN support (requires SSL)
```
Comment (swalkinshaw) [inline README.md]: I don't think this is needed. It's an implementation detail.

**Comment by swalkinshaw** [comment]
@ptibbetts at least there's a VMware box too: https://atlas.hashicorp.com/geerlingguy/boxes/ubuntu1604/

I'd be okay dropping Parallels for now if we have to.

**Comment by swalkinshaw** [comment]
Couple things:

1. We should probably wait until Vagrant 1.8.5 is out to switch to 16.04. 1.8.4 contains fixes necessary for Ubuntu Xenial but it also has NFS bugs.
2. We should use https://atlas.hashicorp.com/bento/boxes/ubuntu-16.04 as the base boxes. Vagrant recommends these as the most official/canonical ones. And they have all 3 providers.

**Comment by ptibbetts** [comment]
Thanks both.

@fullyint it was late and I wasn't sure if I was doing something wrong but the remote user role fails for me when I add cli_options to the raw command. I've [added my log to the commit](https://github.com/fullyint/trellis/commit/49993b5f3db42167af94caf7dcec698f59fb6344#commitcomment-18050484) on your branch. I tested it on a droplet from DO whereas you mentioned AWS, could that have *anything* to do with it?

There was no real reason I chose `whoami` other than needing a command to give to the raw module and it seemed fitting. I'm just about to give it a test (using AWS instead of DO) but is this a good use-case for [failed_when](http://docs.ansible.com/ansible/playbooks_error_handling.html#controlling-what-defines-failure)? Either way I'm happy using `('root' in root_status.stdout_lines)` if you both are (I have very little experience with AWS).

I've started a checklist in the first comment ✔

**Comment by fullyint** [comment]
@ptibbetts thanks for the log. I responded [over there](https://github.com/fullyint/trellis/commit/49993b5f3db42167af94caf7dcec698f59fb6344#commitcomment-18051785). I think the different results we see are attributable to the different cli_options we tested and not attributable to different hosts. I tested both DO and AWS.

The `whoami` seems fitting. Before `ping` it was [`echo can_connect`](https://github.com/roots/trellis/commit/667630288c80afced00572d4a693f356e13222f5#diff-db0d022f251967eca56175a8e2023698). Whatever is consistent across hosts to
- diagnose the `root` user's ability to connect, and
- yield consistent stdout that we can capture in a registered variable and evaluate (given that it seems we can't just rely on the return code from connections to AWS).

I think we need to retain the [`failed_when: false`](https://github.com/roots/trellis/blob/263d05561f1aa0b61ba5162cb71d752cdba54fb3/roles/remote-user/tasks/main.yml#L4). As I understand it, if we ever let the ansible **task** with the whoami fail (as distinct from the whoami **command** inside the task), the playbook will halt. Of course, we don't want the playbook to halt; we just want to test whether the `root` user can connect.

The conditional `('root' in root_status.stdout_lines)` works on DO and AWS but maybe someone with more experience will suggest that we can't trust the `whoami` stdout to be consistent on other hosts.

**Comment by fullyint** [comment]
@ptibbetts thank you for helping me see how the `--ask-pass` option was not working with `cli_options`. I think I've arrived at an acceptable solution ([discussion](https://github.com/fullyint/trellis/commit/49993b5f3db42167af94caf7dcec698f59fb6344#commitcomment-18056215) and [code](https://github.com/fullyint/trellis/commit/c6ac0d2a0ea6e1899550b07b19309684ea49714b)).

**Comment by swalkinshaw** [comment]
Vagrant 1.8.5 is out: https://releases.hashicorp.com/vagrant/1.8.5/

**Comment by ptibbetts** [comment]
https://github.com/roots/trellis/pull/604/commits/c4a161788a65dab8c618d8cac9bf4b11c2d8493a switches to the `bento/ubuntu-16.04` box and updates Vagrant requirements to `1.8.5` 

The symlink for `current` isn't working so the site doesn't get installed

```
failed: [default] (item=example.com) =>
 {"changed": true, "cmd": ["composer", "install"], "delta": "0:00:00.128909", "end": "2016-07-24 18:30:49.013855", "failed": true, "item": "example.com", "rc": 1, "start": "2016-07-24 18:30:48.884946", "stderr":
"You are running composer with xdebug enabled. This has a major impact on runtime performance. See https://getcomposer.org/xdebug
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Composer could not find a composer.json file in /srv/www/example.com/current
To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ \"Getting Started\" section", "stdout": "", "stdout_lines": [], "warnings": []}
```

```
vagrant@example:~$ ls /srv/www/example.com/current/
web
vagrant@example:~$ ls /srv/www/example.com/current/web/
vagrant@example:~$
```

Switching from `nfs` to `rsync` (https://github.com/roots/trellis/pull/604/commits/9d96d54e2078198359f05b0c5c71965b4486ed9a) completes the site installation so there's an issue with NFS and `vm.synced_folder` 

I'll keep looking into it

**Comment by swalkinshaw** [comment]
It looks like the `ubuntu/xenial64` box has been updated and it all works.

I think we should just switch to it for now. We'll worry about finding and testing VMWare and Parallels boxes after.

**Comment by swalkinshaw** [comment]
Here's the full diff for what I got working:

```diff
diff --git a/Vagrantfile b/Vagrantfile
index 08f3b3e..a5e1b50 100644
--- a/Vagrantfile
+++ b/Vagrantfile
@@ -34,7 +34,7 @@ if !Dir.exists?(ENV['ANSIBLE_ROLES_PATH']) && !Vagrant::Util::Platform.windows?
   fail_with_message "You are missing the required Ansible Galaxy roles, please install them with this command:\nansible-galaxy install -r requirements.yml"
 end
 
-Vagrant.require_version '>= 1.5.1'
+Vagrant.require_version '>= 1.8.5'
 
 Vagrant.configure('2') do |config|
   config.vm.box = 'ubuntu/trusty64'
diff --git a/roles/remote-user/tasks/main.yml b/roles/remote-user/tasks/main.yml
index 440ad17..7a95686 100644
--- a/roles/remote-user/tasks/main.yml
+++ b/roles/remote-user/tasks/main.yml
@@ -7,16 +7,15 @@
         ansible-playbook server.yml -e env={{ env }} -u root --ask-pass
     when: cli_ask_pass | default(false)
 
-    local_action: command ansible {{ inventory_hostname }} -m ping -u root {{ cli_options | default('') }}
+    local_action: command ansible {{ inventory_hostname }} -m raw -a whoami -u {{ ansible_user | default('root') }} {{ cli_options | default('') }}
     failed_when: false
     changed_when: false
     register: root_status
 
   - name: Set remote user for each host
     set_fact:
-      ansible_user: "{{ root_status | default({'failed':false}) | success | ternary('root', admin_user) }}"
+      ansible_user: "{{ ('root' in root_status.stdout_lines) | ternary(ansible_user | default('root'), admin_user) }}"
 
   when: ansible_user is not defined
 
diff --git a/server.yml b/server.yml
index e422a67..9f27637 100644
--- a/server.yml
+++ b/server.yml
@@ -9,6 +9,14 @@
   roles:
     - { role: remote-user, tags: [remote-user, always] }
 
+- name: Install prerequisites
+  hosts: web:&{{ env }}
+  gather_facts: false
+  become: yes
+  pre_tasks:
+    - name: Install Python 2.x
+      raw: sudo apt-get -y install python-simplejson
+
 - name: WordPress Server - Install LEMP Stack with PHP 7.0 and MariaDB MySQL
   hosts: web:&{{ env }}
   become: yes
```

`sudo apt-get -y install python-simplejson` is what Ansible itself recommends. The symlink wasn't needed either.

**Comment by swalkinshaw** [comment]
My testing all worked with the `vagrant` user. That was an older bug in the box so make sure you run `vagrant box update` to get the latest.

**Comment by ptibbetts** [comment]
I'm on the latest box 

```
vagrant box update
 ==> default: Box 'ubuntu/xenial64' (v20160725.0.0) is running the latest version
```

Running `vagrant up` works until

```
==> default: Bindfs seems to not be installed on the virtual machine, installing now
==> default: Creating bind mounts for selected devices
==> default: Cannot bind to user 'vagrant' because it doesn't exist
==> default: Updating /etc/hosts file on active guest machines...
==> default: Updating /etc/hosts file on host machine (password may be required)...
Password:
==> default: Running provisioner: ansible...
    default: Running ansible-playbook...
```

however the script carries onto the end and displays

```
TASK [wordpress-setup : Create web root] ***************************************
System info:
  Ansible 2.0.2.0; Darwin
  Trellis at "Upgraded Ubuntu from 14.04 Trusty to 16.04 Xenial"
---------------------------------------------------
chown failed: failed to look up user vagrant
failed: [default] (item=example.com) => {"failed": true, "gid": 0, "group": "root", "item": "example.com", "mode": "0755", "owner": "root", "path": "/srv/www", "size": 4096, "state": "directory", "uid": 0}
```

```
PLAY RECAP *********************************************************************
default                    : ok=91   changed=66   unreachable=0    failed=1

Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
```

😕

**Comment by ptibbetts** [comment]
It looks like the `ubuntu/xenial64` box is missing the vagrant user. I've switched to `geerlingguy/ubuntu1604` again and it's working fine with the vagrant user (https://github.com/roots/trellis/pull/604/commits/828e48a64232724396c47d12a8b0dca60f77ed25).

Having updated to VirtualBox from `5.0.22` to `5.0.26` and pulled from master the `bento/ubuntu-16.04` box is working without any issues for me (https://github.com/roots/trellis/pull/604/commits/40c0ce3106fc4dc13d325312af9eb7187264bc8b).

I could

- [ ] tidy this up and use the `geerlingguy` box
- [ ] tidy this up and use the `bento/ubuntu-16.04` (has parallels and vmware boxes)
- [ ] add the vagrant user to the `ubuntu/xenial64` user (somehow)

My preference is the bento box

**Comment by swalkinshaw** [comment]
I just tried bento and it's working as well. Let's stick with it 👍

**Comment by swalkinshaw** [comment]
Looks like this needs a rebase from master as well. Some of the changes are already in there.

**Comment by swalkinshaw** [comment]
This PR/branch has become a bit of a mess with all the changes.

@ptibbetts I'm closing it in favour of https://github.com/roots/trellis/pull/626

It's all the necessary changes simplified (you remain as author 😄 )

**Comment by ptibbetts** [comment]
Thanks @swalkinshaw, I was also making a mess of rebasing it last night 😅  so I pushed what I had with the intention of trying again today

**Pull request closed without merging**